src/main/java/com/opinta/service/DiscountPerCounterpartyServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+
+@Service
+@Slf4j
+public class DiscountPerCounterpartyServiceImpl implements DiscountPerCounterpartyService {
+    private final DiscountPerCounterpartyDao discountPerCounterpartyDao;
+    private final DiscountPerCounterpartyMapper discountPerCounterpartyMapper;
+    private final CounterpartyService counterpartyService;
+    private final DiscountService discountService;
+    private final UserService userService;
+    private final Format dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+
+    @Autowired
+    public DiscountPerCounterpartyServiceImpl(DiscountPerCounterpartyDao discountPerCounterpartyDao,
+                                              DiscountPerCounterpartyMapper discountPerCounterpartyMapper,
+                                              CounterpartyService counterpartyService,
+                                              DiscountService discountService, UserService userService) {
+        this.discountPerCounterpartyDao = discountPerCounterpartyDao;
+        this.discountPerCounterpartyMapper = discountPerCounterpartyMapper;
+        this.counterpartyService = counterpartyService;
+        this.discountService = discountService;
+        this.userService = userService;
+    }
+
+    @Override
+    @Transactional
+    public List<DiscountPerCounterparty> getAllEntities(User user) {
+        log.info(LogMessageUtil.getAllLogEndpoint(DiscountPerCounterparty.class));
+        return discountPerCounterpartyDao.getAll(user);
+    }
+
+    @Override
+    @Transactional
+    public DiscountPerCounterparty getEntityByUuid(UUID uuid, User user) throws AuthException,
+            IncorrectInputDataException {
+        log.info(LogMessageUtil.getByIdLogEndpoint(DiscountPerCounterparty.class, uuid));
+        DiscountPerCounterparty discountPerCounterparty = discountPerCounterpartyDao.getByUuid(uuid);
+        if (discountPerCounterparty == null) {
+            log.error(getByIdOnErrorLogEndpoint(DiscountPerCounterparty.class, uuid));
+            throw new IncorrectInputDataException(getByIdOnErrorLogEndpoint(DiscountPerCounterparty.class, uuid));
+        }
+
+        userService.authorizeForAction(discountPerCounterparty, user);
+
+        return discountPerCounterpartyDao.getByUuid(uuid);
+    }
+
+    @Override
+    @Transactional
+    public DiscountPerCounterparty saveEntity(DiscountPerCounterparty discountPerCounterparty, User user)
+            throws AuthException, IncorrectInputDataException, PerformProcessFailedException {
+        Counterparty counterparty = counterpartyService.getEntityByUuid(
+                discountPerCounterparty.getCounterparty().getUuid(), user);
+        Discount discount = discountService.getEntityByUuid(discountPerCounterparty.getDiscount().getUuid());
+
+        discountPerCounterparty.setCounterparty(counterparty);
+        discountPerCounterparty.setDiscount(discount);
+        validateDiscountTime(discountPerCounterparty);
```

Comment:
Plz, rename to validateDiscountPerCounterparty and put to DiscountPerCounterparty.java
Delete local methods for discount validation